pkg/querier/querier.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package querier
+
+import (
+	"github.com/vmware-tanzu/antrea/pkg/apis/clusterinformation/v1beta1"
+	networkingv1beta1 "github.com/vmware-tanzu/antrea/pkg/apis/networking/v1beta1"
+)
+
+type AgentQuerier interface {
+	GetAgentInfo() *v1beta1.AntreaAgentInfo
+}
+
+type ControllerQuerier interface {
+	GetControllerInfo() *v1beta1.AntreaControllerInfo
+}
+
+type NetworkPolicyInfoQuerier interface {
+	GetNetworkPolicyNum() int
+	GetAddressGroupNum() int
+	GetAppliedToGroupNum() int
+	GetNetworkPolicyControllerInfo() v1beta1.NetworkPolicyControllerInfo
+}
+
+type AgentNetworkPolicyInfoQuerier interface {
+	NetworkPolicyInfoQuerier
+	GetControllerConnectionStatus() bool
```

Comment:
Even this one is not really relevant to NetworkPolicy, but now I got it is not your change and we put it to NetworkPolicy because we got the data from NetworkPolicyController. Then I do not have a strong opinion.

Reply:
I have refactored the code.

* pkg/querier
  * interface NetworkPolicyInfoQuerier and related
  * util funcs: GetSelfPod(), GetSelfNode(), etc.
* pkg/agent/querier
  * struct agentQuerier: fields like nodeName, nodeSubnet, etc.
  * interface AgentQuerier:  all agent query funcs
* pkg/controller/querier
  * struct controllerQuerier: fields networkPolicyInfoQuerier
  * struct ControllerQuerier: all controller query funcs
* pkg/monitor/agent
  * struct agentMonitor: client and interface AgentQuerier
  * agent CRD related funcs
* pkg/monitor/controller
  * struct controllerMonitor: client, nodeInformer, etc. and ControllerQuerier
  * controller CRD related funcs

Please have a review @jianjuns .